### ISSUE TITLE
Update Google Analytics tracking before 2023

### ISSUE THREAD

**Issue by eporter23** [emory-libraries/dlp-curate#1803]
The older universal analytics method is being closed out on 7/1/23.

https://support.google.com/analytics/answer/11583528

Current reporting metrics to recreate:

- Top 25 pages ([test example](https://analytics.google.com/analytics/web/#/analysis/p346689597/edit/9v3GzSIvQomGTrO1AEjXKQ))
- Top 25 Collections ([test example](https://analytics.google.com/analytics/web/#/analysis/p346689597/edit/8aWmujV4RXqhrU-PEa2Ynw))
- Facet clicks ([test example](https://analytics.google.com/analytics/web/#/analysis/p346689597/edit/-FDzjpuLRpCnW2Vdw8ZDWg))
- Facet explorations [(test example](https://analytics.google.com/analytics/web/#/analysis/p346689597/edit/lOv5ZE35TVed5PkZcCMdhA))
- Views by Format ([test example](https://analytics.google.com/analytics/web/#/analysis/p346689597/edit/yjb_YOuZTaienFwZ1FEyYA))
- Search terms ([test example](https://analytics.google.com/analytics/web/#/analysis/p346689597/edit/0u7XcyMwSOuc60j1JbpKNA))
- Search field/categorization ([test example](https://analytics.google.com/analytics/web/#/analysis/p346689597/edit/P_nm4V9HQ2uRFRinZfls1w) plus [advanced search](https://analytics.google.com/analytics/web/#/analysis/p346689597/edit/bE4ewEfXSFq7DAcFOibdmg))
- Top 25 pages per each Library ([test example](https://analytics.google.com/analytics/web/#/analysis/p346689597/edit/eWgFofmQTIGQJmYlAKtm6g))
- Summary dashboard ([test example](https://analytics.google.com/analytics/web/#/p346689597/assetlibrary/dashboard/edit?params=_u..nav%3Dmaui&r=4409346086))

Suggested tasks:

- [x] Research if a new code is needed for Lux (no, because it was a Universal Analytics account)
- [x] Add new code if needed (not needed for DLP)
- [x] Test new code in test
- [x] Test if our custom event tracking still works (some events needed to be defined in the GA configuration, but original scripts still work)
- [x] Confirm if saved reports must be recreated (yes)
- [x] Confirm if site search terms are tracked (yes)
- [x] Recreate search use categorization
- [x] Identify other remediations needed - custom dimensions needed to be defined in some cases. See also the Working Notes document below.
- [x] Recreate saved reports
- [x] Implement updates in test
- [x] Implement updates in production

Additional notes/references
https://support.google.com/analytics/answer/10759417?hl=en&ref_topic=10737980
https://omnitail.net/universal-analytics-vs-google-analytics-4/
https://www.lovesdata.com/blog/google-analytics-4-events

Working notes
https://docs.google.com/document/d/1cduB81wmojrwxjiLU33b23Prh5ELmFSRec0qUaSdDMk/edit?usp=sharing

**Comment by eporter23** [comment]
I went through the conversion wizard for our Google Analytics account for Lux. At first pass, it seems that Google can covert the property for us without requiring us to update the code in our application.

**Comment by eporter23** [comment]
Per Google's documentation, if our existing GA code in applications/websites use the `gtag.js `script, we should be able to update our property without having to replace the script tags.

**Comment by eporter23** [comment]
After 48 hours the tracking does seem to be active. Our custom events also seem to be tracked, but I have not yet figured out how to categorize specific event types such as our page views by Holding Repository report.

**Comment by eporter23** [comment]
Site search tracking is not yet working but I have tried a few new settings and will monitor if they worked.

**Comment by eporter23** [comment]
As of 1/4/23, I have recreated most functionality. Still in progress: search type categorization (e.g. Title vs Author etc).

**Comment by eporter23** [comment]
1/6/23 - search field usage is now being tracked.

**Comment by eporter23** [comment]
1/10/23 basic dashboard created

**Comment by eporter23** [comment]
GA4 is now running in parallel with UA in production for Lux.

**Comment by eporter23** [comment]
Curate tracking added in production on 3/7/23.
